src/Core/Validation/FieldValidation/DateFieldValidator.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\FieldValidation\FieldValidator;
+use SilverStripe\Core\Validation\ValidationResult;
+
+/**
+ * Validates that a value is a valid date, which means that it follows the equivalent formats:
+ * - PHP date format Y-m-d
+ * - SO format y-MM-dd i.e. DBDate::ISO_DATE
+ * Emtpy values are allowed
```

Comment:
```suggestion
 * Empty values are allowed
```
Probably mention "empty values are allowed" for _all_ validators where that's true, or else add a "empty values are allowed unless otherwise stated" to `FieldValidator` and then explicitly state which are _not_ allowed empty values.

Reply:
Updated to say that it's blank strings specifically